Extension/src/LanguageServer/settings.ts

```diff
@@ -312,89 +309,186 @@ export class CppSettings extends Settings {
+        }
+        return false;
+    }
+
+    // This helper function validates whether the given value is a valid enum value for the given setting.
+    private isValidEnum(settingName: string, value: any): boolean {
+        const setting = getRawSetting("C_Cpp." + settingName);
+        if (this.validateEnum(setting.enum, value)) {
+            return value;
+        }
+        return setting.default;
+    }
+
+    private isArrayOfCommentContinuationPatterns(x: any): x is (string | CommentPattern)[] {
+        return isArray(x) && x.every(y => isString(y) || this.isCommentPattern(y));
+    }
+
+    private isCommentPattern(x: any): x is CommentPattern {
+        return isString(x.begin) && isString(x.continue);
+    }
+
+    public get maxConcurrentThreads(): number | null { return this.getAsNumber("maxConcurrentThreads"); }
+    public get maxMemory(): number | null { return this.getAsNumber("maxMemory"); }
+    public get maxSymbolSearchResults(): number { return this.getAsNumber("maxSymbolSearchResults"); }
+    public get maxCachedProcesses(): number { return this.getAsNumber("maxCachedProcesses"); }
+    public get intelliSenseMaxCachedProcesses(): number | null { return this.getAsNumber("intelliSense.maxCachedProcesses"); }
+    public get intelliSenseMaxMemory(): number | null { return this.getAsNumber("intelliSense.maxMemory"); }
+    public get referencesMaxConcurrentThreads(): number | null { return this.getAsNumber("references.maxConcurrentThreads"); }
+    public get referencesMaxCachedProcesses(): number { return this.getAsNumber("references.maxCachedProcesses"); }
+    public get referencesMaxMemory(): number | null { return this.getAsNumber("references.maxMemory"); }
+    public get codeAnalysisMaxConcurrentThreads(): number | null { return this.getAsNumber("codeAnalysis.maxConcurrentThreads"); }
+    public get codeAnalysisMaxMemory(): number | null { return this.getAsNumber("codeAnalysis.maxMemory"); }
+    public get codeAnalysisUpdateDelay(): number { return this.getAsNumber("codeAnalysis.updateDelay"); }
+    public get codeAnalysisExclude(): Excludes { return this.getAsKeyValueObject("codeAnalysis.exclude", "string", "boolean"); }
+    public get codeAnalysisRunAutomatically(): boolean { return this.getAsBoolean("codeAnalysis.runAutomatically"); }
+    public get codeAnalysisRunOnBuild(): boolean { return this.getAsBoolean("codeAnalysis.runOnBuild"); }
+    public get clangTidyEnabled(): boolean { return this.getAsBoolean("codeAnalysis.clangTidy.enabled"); }
+    public get clangTidyConfig(): string { return this.getAsString("codeAnalysis.clangTidy.config"); }
+    public get clangTidyFallbackConfig(): string { return this.getAsString("codeAnalysis.clangTidy.fallbackConfig"); }
+    public get clangTidyHeaderFilter(): string | null { return this.getAsString("codeAnalysis.clangTidy.headerFilter"); }
+    public get clangTidyArgs(): string[] { return this.getAsArrayOfStrings("codeAnalysis.clangTidy.args"); }
```

Comment:
We have a similar problem for `getAsString` when `null` is an acceptable output.